README.md

```diff
@@ -1 +1,419 @@
+  | Post String Int
+  | Feed { search :: Maybe String, sorting :: Maybe Sort }
+
+derive instance genericRoute :: Generic Route _
+
+route :: RouteDuplex' Route
+route = root $ sum
+  { ...
+  , "Feed": path "feed" $ params 
+      { search: optional <<< string 
+      , sorting: optional <<< sort
+      }
+  }
+```
+
+Next, lets make our `Route` data type better by providing newtypes to uniquely identify a string as a `Username` or an int as a `PostId`.
+
+```purescript
+newtype Username = Username String
+derive instance newtypeUsername :: Newtype Username _
+
+newtype PostId = PostId Int
+derive instance newtypePostId :: Newtype PostId _
+
+data Route
+  = Root
+  | Profile Username
+  | Post Username PostId
+  | Feed { search :: Maybe String, sorting :: Maybe Sort }
+```
+
+We could write new combinators as we did for the `Sort` data type, but we don’t really have brand-new data types here. We have newtypes around a `String` and `Int`, which already have combinators available, so we ought to re-use them. This re-use is trivial if we use the `_Newtype` iso from `purescript-profunctor-lenses` along with the existing combinators to create our two new codecs:
+
+```purescript
+uname :: RouteDuplex' Username
+uname = _Newtype segment
+
+-- re-use the `int` combinator
+postId :: RouteDuplex' PostId
+postId = _Newtype (int segment)
+```
+
+Finally, we can replace our earlier segments with these new codecs.
+
+```purescript
+route :: RouteDuplex' Route
+route = root $ sum
+  { "Root": noArgs
+  , "Profile": "user" / uname
+  , "Post": "user" / uname / "post" / postId
+  , "Feed": "feed" / params 
+      { search: optional <<< string 
+      , sorting: optional <<< sort
+      }
+  }
+```
+
+## Example: Composing codecs to represent CRUD operations
+We’ve seen the `RouteDuplex’ a` type all over the place, whether to represent a small codec for integers or strings or a larger one for our complex sum type. We can create codecs of any size and compose them into larger structures. Let’s walk through an example by extending our routing data type to accommodate create, read, and update operations for posts in our system.
+
+First, we’ll define a data type to represent creating, reading, and updating a resource dependent on some kind of identifier, `a`:
+
+```purescript
+data CRU a
+  = Create
+  | Read a
+  | Update a
+
+derive instance genericCRU :: Generic (CRU a) _
+```
+
+Next, we’ll again use the `sum` function to write a codec for this sum type. We don’t know how to handle `a`, so we’ll accept a codec to handle it as an argument. We’d like to handle three cases:
+
+* `/` should represent creation
+* `/:id` should represent reading
+* `/edit/:id` should represent updating
+
+Exactly the same way we wrote a codec for our `Route` type we can write one for our new `CRU` type:
+
+```purescript
+cru :: forall a. RouteDuplex' a -> RouteDuplex' (CRU a)
+cru inner = sum
+  { "Create": noArgs
+  , "Read": inner
+  , "Update": path "edit" inner
```

Comment:
Should this use `"edit" / inner` syntax?

Reply:
👍

Reply:
If `suffix` is ready, then I could change this to be `inner / suffix "edit"` or whatever correct usage is of that function -- that at least shows off the possibility.